app/src/org/commcare/models/database/user/models/EntityStorageCache.java

```diff
@@ -130,4 +132,32 @@ public static int getSortFieldIdFromCacheKey(String detailId, String cacheKey) {
             return -1;
         }
     }
+
+    public static void tryWipeCache() {
```

Comment:
Is the expected behavior of this function either that it wipes the cache immediately *or* that it triggers a setting that will required the cache to wipe at a later time?

Right now i'm not comfortable with the guaranteed execution of that process given the Session exception path here. Let's chat over voice on that.